include/ocpn_plugin.h

```diff
@@ -6305,4 +6307,390 @@ extern DECL_EXP void EnableTenHertzUpdate(bool enable);
  */
 extern DECL_EXP void ConfigFlushAndReload();
 
+/**
+ * Contains information about a single navigation or safety violation in a
+ * route.
+ */
+class DECL_EXP PlugIn_NavigationViolation {
+public:
+  PlugIn_NavigationViolation();
+  ~PlugIn_NavigationViolation();
+
+  int m_SegmentIndex;   //!< Index of waypoint where violation occurred.
+  int m_ViolationType;  //!< Safety type bitmask of PI_NavigationType indicating
+                        //!< what was violated.
+  double m_ViolationDistance;  //!< Distance to nearest safety constraint (nm)
+  double m_ClosestLat;         //!< Latitude of closest constraint point
+  double m_ClosestLon;         //!< Longitude of closest constraint point
```

Comment:
TODO: need references to marks, obstructions, objects... This could be used for visualization.